src/nidigital/examples/nidigital_burst_with_start_trigger.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/python
+
+import argparse
+import nidigital
+import os
+import sys
+
+
+def example(resource_name, options, trigger_source=None, trigger_edge=None):
+
+    with nidigital.Session(resource_name=resource_name, options=options) as session:
+
+        dir = os.path.join(os.path.dirname(__file__), 'burst_with_start_trigger_files')
+
+        # Load the pin map (.pinmap) created using the Digital Pattern Editor
+        pin_map_filename = os.path.join(dir, 'PinMap.pinmap')
+        session.load_pin_map(pin_map_filename)
+
+        # Load the specifications (.specs), levels (.digilevels), and timing (.digitiming) sheets created using the Digital Pattern Editor
+        spec_filename = os.path.join(dir, 'Specifications.specs')
+        levels_filename = os.path.join(dir, 'PinLevels.digilevels')
+        timing_filename = os.path.join(dir, 'Timing.digitiming')
+        session.load_specifications_levels_and_timing(spec_filename, levels_filename, timing_filename)
+
+        # Apply the settings from the levels and timing sheets we just loaded to the session
+        session.apply_levels_and_timing(levels_filename, timing_filename)
+
+        # Loading the pattern file (.digipat) created using the Digital Pattern Editor
+        pattern_filename = os.path.join(dir, 'Pattern.digipat')
+        session.load_pattern(pattern_filename)
+
+        if trigger_source is not None:
+            # Specify a source and edge for the external start trigger
+            session.start_trigger_type = nidigital.TriggerType.DIGITAL_EDGE
+            session.digital_edge_start_trigger_source = trigger_source
+            session.digital_edge_start_trigger_edge = nidigital.DigitalEdge.RISING if trigger_edge == 'Rising' else nidigital.DigitalEdge.FALLING
+
+        # If start trigger is configured, waiting for the trigger to start bursting and then blocks until the pattern is done bursting
+        # Else just start bursting and block until the pattern is done bursting
+        session.burst_pattern(start_label='new_pattern')
+        if trigger_source is None:
+            print('Start bursting pattern')
+        else:
+            print('Wait for start trigger and then start bursting pattern')
+
+        # Disconnect all channels using programmable onboard switching
+        session.selected_function = nidigital.SelectedFunction.DISCONNECT
```

Comment:
I am no NI-Digital expert so bear with me...
```
        session.burst_pattern(start_label='new_pattern')
```
This is what makes the hardware begin generation.
But we don't wait for the pattern to finish before closing the session and printing `Done bursting pattern`. Shouldn't we wait?

Reply:
[Link to burst_pattern documentation](https://nimi-python.readthedocs.io/en/master/nidigital/class.html#nidigital.Session.burst_pattern)

There is an optional parameter wait_until_done that is set to true by default and will block until the pattern is done.